discovery_engine/test/document_manager_test.dart

```diff
@@ -116,13 +131,17 @@ Future<void> main() async {
         () => mgr.updateUserReaction(id3, UserReaction.positive),
         throwsArgumentError,
       );
+      expect(changedDocsReporter.changedDocuments, emitsDone);
+      await changedDocsReporter.close();
```

Comment:
Is that necessary given that we do that anyway in `tearDown`.

Reply:
I'll take another look, but it might be, closing the underlying stream controller is necessary to finish this test, and if we only do it the `tearDown` the test will timeout

Reply:
ok, I've checked and we need to close the stream, otherwise the test will timeout.